apt/private/index.bzl

```diff
@@ -85,17 +85,19 @@ def _deb_package_index_impl(rctx):
             ),
         )
 
+        deps = depset([
```

Comment:
We had packages that shared duplicate transitive dependencies. The duplicate deps were a blocking issue, so I used a depset to dedupe packages.

Reply:
Do you have an example the package(s) that broke? Were they packages in the NVIDIA CUDA repos?

Reply:
@jjmaestro I had this in my sources:

```
  - channel: nvidia cuda
    url: https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2004/x86_64
```

And this in my packages:

```
  # CUDA Dependencies
  - cuda-11-8
  - cudnn9-cuda-11-8
```

I think those would replicate the issue.

Reply:
@benmccown OK, I looked more carefully and I was using the wrong `Packages` index / arch but that's also the problem for the repro, I can't get both of those packages for both `amd64` and `arm64` (they are only available for `amd64` IIRC).

Have a look at https://github.com/jjmaestro/rules_distroless/tree/testing-cuda-packages, I have one commit there so far: https://github.com/jjmaestro/rules_distroless/commit/f216178ce98ba1d21c8759818b2aa3949e37cc92, this is a test running on top of https://github.com/GoogleContainerTools/rules_distroless/pull/97.

I've gotten it to work with just the `cuda-11-8` package in `arm64`, and since that's the largest package by far, I think it's probably working but I'll change the example I'm working on to only `amd64` and see if something breaks when adding both packages.

Reply:
I've found yet-another-issue when testing with `amd64`... the failure I'm getting is because there's a Debian package that's already compressed as a `tar.gz`, see #98. I'll see if I can work around that issue and test things further but so far, I haven't seen duplicate transitive dependencies.

Reply:
Alright, @benmccown please check [jjmaestro/rules_distroless/tree/testing-cuda-packages](https://github.com/jjmaestro/rules_distroless/tree/testing-cuda-packages).

I didn't get any errors / issues with duplicate transitive dependencies. I have it all working after porting #99 and also making sure I flatten the layers to avoid `max depth exceeded` (#36) and also, using `pkg_tar` to flatten, to avoid `duplicates of file paths not supported` Docker errors (e.g. https://github.com/bazelbuild/rules_docker/issues/246).

Can you check that branch and confirm if that works for you?

Thanks!